src/webgpu/shader/execution/expression/call/builtin/textureStore.spec.ts

```diff
@@ -121,6 +128,265 @@ Parameters:
+  .params(u => u.combineWithParams([...TexelFormats, { format: 'bgra8unorm', _shaderType: 'f32' }]))
+  .beforeAllSubcases(t => {
+    if (t.params.format === 'bgra8unorm') {
+      t.selectDeviceOrSkipTestCase('bgra8unorm-storage');
+    } else {
+      t.skipIfTextureFormatNotUsableAsStorageTexture(t.params.format as GPUTextureFormat);
+    }
+  })
+  .fn(t => {
+    const { format, _shaderType } = t.params;
+    const values = inputArray(format);
+    let wgsl = `
+const range = array(`;
+    for (const v of values) {
+      wgsl += `${v},\n`;
+    }
+
+    wgsl += `
+);
+
+@group(0) @binding(0)
+var tex : texture_storage_1d<${format}, write>;
+
+@compute @workgroup_size(${values.length})
+fn main(@builtin(global_invocation_id) gid : vec3u) {
+  let idx = gid.x;
+  let scalarVal = range[idx];
+  let vecVal = vec4(scalarVal);
+  let val = vec4<${_shaderType}>(vecVal);
+  textureStore(tex, gid.x, val);
+}
+`;
+
+    const numTexels = values.length;
+    const textureSize: GPUExtent3D = { width: numTexels, height: 1, depthOrArrayLayers: 1 };
+    const texture = t.createTextureTracked({
+      format: format as GPUTextureFormat,
+      dimension: '1d',
+      size: textureSize,
+      mipLevelCount: 1,
+      usage: GPUTextureUsage.STORAGE_BINDING | GPUTextureUsage.COPY_SRC,
+    });
+
+    const pipeline = t.device.createComputePipeline({
+      layout: 'auto',
+      compute: {
+        module: t.device.createShaderModule({
+          code: wgsl,
+        }),
+        entryPoint: 'main',
+      },
+    });
+    const bg = t.device.createBindGroup({
+      layout: pipeline.getBindGroupLayout(0),
+      entries: [
+        {
+          binding: 0,
+          resource: texture.createView({
+            format: format as GPUTextureFormat,
+            dimension: '1d',
+          }),
+        },
+      ],
+    });
+
+    const encoder = t.device.createCommandEncoder();
+    const pass = encoder.beginComputePass();
+    pass.setPipeline(pipeline);
+    pass.setBindGroup(0, bg);
+    pass.dispatchWorkgroups(1, 1, 1);
+    pass.end();
+    t.queue.submit([encoder.finish()]);
+
+    let bytesPerTexel = 4;
+    switch (format) {
+      case 'rgba16uint':
+      case 'rgba16sint':
+      case 'rgba16float':
+      case 'rg32uint':
+      case 'rg32sint':
+      case 'rg32float':
+        bytesPerTexel = 8;
+        break;
+      case 'rgba32uint':
+      case 'rgba32sint':
+      case 'rgba32float':
+        bytesPerTexel = 16;
+        break;
+      default:
+        break;
+    }
+
+    const buffer = t.copyWholeTextureToNewBufferSimple(texture, 0);
+    const uintsPerTexel = bytesPerTexel / 4;
+    const expected = new Uint32Array([
+      ...iterRange(numTexels * uintsPerTexel, x => {
+        const idx = Math.floor(x / uintsPerTexel);
+        const shaderVal = values[idx];
+        switch (format) {
+          case 'rgba8unorm':
+          case 'bgra8unorm':
+            return pack4x8unorm(shaderVal, shaderVal, shaderVal, shaderVal);
```

Comment:
This test splats the scalar across the whole vector, so it can't detect the case where the channel ordering is messed up.

Please modify the test to detect channel ordering bugs.

E.g. for a k-channel format, write k outputs per invocation, where on the kth value the kth component is set to 0.

Reply:
This is done is `bgra8unorm_swizzle`. Do you think it is necessary for other formats?

Reply:
Yeah, I think the other formats should have it too.

Reply:
Added that in the latest commit.